heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -513,6 +526,11 @@ export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
         }
       }
 
+      // unset the NODE_ENV only if it's test
+      if (process.env.NODE_ENV === 'test') {
```

Comment:
Is correct to unset an environment variable that the user had explicitly set?  Maybe Heft should unset `NODE_ENV` if-and-only-if Heft itself set `NODE_ENV` earlier in this function?

Reply:
@octogonz 

> Is correct to unset an environment variable that the user had explicitly set? 

Probably not, but that's the thing. ATM we don't know if the user is the one who set it or not. 

> Heft should unset NODE_ENV if-and-only-if Heft itself set NODE_ENV earlier in this function?

I thought of that initially then checked how jest does and implemented the change in the `spirit of jest`.  

I would suggest implementing this in the simplest form possible that is compatible with jest. This means to set the env var and not unset it then maybe do it the `right way` and combine env vars as written here https://github.com/microsoft/rushstack/issues/2420#issuecomment-756908991

I have never seen anyone using the NODE_ENV=test in any other way or purpose than for testing. 

Just as a side note, I disagree with unsettling the variable if it's already set.